x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InternalInferModelAction.java

```diff
@@ -72,18 +67,7 @@ public Request(StreamInput in) throws IOException {
             super(in);
             this.modelId = in.readString();
             this.objectsToInfer = Collections.unmodifiableList(in.readList(StreamInput::readMap));
-            if (in.getVersion().onOrAfter(Version.V_7_8_0)) {
-                this.update = in.readNamedWriteable(InferenceConfigUpdate.class);
-            } else {
-                InferenceConfig oldConfig = in.readNamedWriteable(InferenceConfig.class);
-                if (oldConfig instanceof RegressionConfig) {
-                    this.update = RegressionConfigUpdate.fromConfig((RegressionConfig)oldConfig);
-                } else if (oldConfig instanceof ClassificationConfig) {
-                    this.update = ClassificationConfigUpdate.fromConfig((ClassificationConfig) oldConfig);
-                } else {
-                    throw new IOException("Unexpected configuration type [" + oldConfig.getName() + "]");
-                }
-            }
```

Comment:
As above, this is very complex and it would be useful for anyone changing it on the master branch to be aware the complexity exists in 7.x.  So it's probably best to leave this file untouched until closer to 8.0 feature freeze too.